packages/esm-patient-chart-app/src/visit/visit-prompt/retrospective-dialog.component.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { useCallback, useMemo } from 'react';
+import {
+  Button,
+  ModalBody,
+  ModalHeader,
+  ModalFooter,
+  DatePicker,
+  DatePickerInput,
+  Stack,
+  Row,
+  Column,
+  Form,
+  TimePicker,
+  TimePickerSelect,
+  SelectItem,
+} from '@carbon/react';
+import styles from './retrospective-dialog.scss';
+import { useTranslation } from 'react-i18next';
+import {
+  NewVisitPayload,
+  formatDate,
+  saveVisit,
+  setCurrentVisit,
+  showNotification,
+  toDateObjectStrict,
+  toOmrsIsoString,
+} from '@openmrs/esm-framework';
+import { Controller, FormProvider, useForm } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import dayjs from 'dayjs';
+import { amPm, convertTime12to24 } from '@openmrs/esm-patient-common-lib';
+
+interface RetrospectiveDialogProp {
+  patientUuid: string;
+  closeModal: () => void;
+  onCancel: () => void;
+  visitDate: string;
+  payload: NewVisitPayload;
+}
+
+const RetrospectiveDialog: React.FC<RetrospectiveDialogProp> = ({
+  closeModal,
+  visitDate,
+  payload,
+  patientUuid,
+  onCancel,
+}) => {
+  const { t } = useTranslation();
+
+  const retrospectiveFormSchema = useMemo(() => {
```

Comment:
Why is this even part of the React object at all? Could we just as easily (and better) define this as a const?